### DISCUSSION TITLE
[next-auth][error][get_user_by_provider_account_id_error] TypeError: prisma[Account].findOne is not a function

### DISCUSSION THREAD

**Discussion by kasorin** [prisma/prisma#5269] [Q&A] [answered]
Hello!
I've been following this (https://vercel.com/guides/nextjs-prisma-postgres) guide to learn prisma application development, and it works fine until step4, but when I press the "sign in with github" button in step5, it throws an error like the one in the title and it doesn't work.

[next-auth][error][get_user_by_provider_account_id_error] TypeError: prisma[Account].findOne is not a function

[next-auth][error][oauth_callback_handler_error] Error: GET_USER_BY_PROVIDER_ACCOUNT_ID_ERROR

When I set the debug option to true, the information of the github account used for authentication was displayed.
The OAuth application settings on github also showed "1 User", so it seems to be working up to that point.

Since I was redirected to "http://localhost:3000/api/auth/signin?error=Callback", I think there is an error somewhere in the callback, but I have no idea how to solve it. Can someone please help me?

package.json
```"dependencies": {
    "@prisma/client": "^2.15.0",
    "next": "^10.0.5",
    "next-auth": "^3.1.0",
    "react": "16.14.0",
    "react-dom": "16.14.0",
    "react-markdown": "5.0.2"
  },
  "devDependencies": {
    "@prisma/cli": "2.15.0",
    "@types/next-auth": "3.1.14",
    "@types/node": "14.14.6",
    "@types/react": "16.9.56",
    "typescript": "4.0.5"
  }
```

.env
```
GITHUB_ID=[secret]
GITHUB_SECRET=[secret]
NEXTAUTH_URL=http://localhost:3000
```

github OAuth aplication settings

![image](https://user-images.githubusercontent.com/26285529/105673288-2d1b6500-5f29-11eb-8609-71ecee4c3dd8.png)

error screen

![image](https://user-images.githubusercontent.com/26285529/105673529-85eafd80-5f29-11eb-8cd6-0ea19bd1bf5e.png)

**Comment by ryands17** [comment, marked as answer]
@kasorin 👋 

This is an issue with the `next-auth` library and you can follow the workaround specified here https://github.com/nextauthjs/next-auth/issues/1191.

**Reply by kasorin** [reply]
@ryands17 Hi!
I read the link you gave me, followed it to the canary version of next-auth, and it worked correctly! Thank you!

**Reply by drelbert** [reply]
Thank you @ryands17  for the link and @kasorin for presenting the issue.  I too had the exact error (code, settings, and error message) and updated to: 3.3.0-canary.3 but the error message persists.  Any additional insights that could be helpful?

**Reply by bhatvikrant** [reply]
Same error for, the workaround also doesn't seem to work

**Reply by vivekDebugs** [reply]
Had similar issue, but worked for me as I updated my packages.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
